### ISSUE TITLE
Crypt using GPG key

### ISSUE THREAD

**Issue by Marthym** [xwmx/nb#52]
Maybe it can be great to encrypt using specified GPG key

This will avoid to retain multiple password

**Comment by xwmx** [comment]
Yeah, I have a lot of ideas around this. I'm aiming to have different encryption and authentication strategies implemented as plugins. The API should be basically there. I like keeping simple, clear file encryption as the baseline in the core. I feel like it should be possible to implement just about anything around encryption, passwords, and keys as plugins with no more than small hook in the [`_encrypt_file()`](https://github.com/xwmx/nb/blob/master/nb#L1247) and [`_decrypt_file()`](https://github.com/xwmx/nb/blob/master/nb#L1021) functions.

**Comment by oliryde** [comment]
With the current implementation wouldn't a plugin need to override the encryption and decryption functions?

Would it make sense to add encryption methods like subcommands are added?
It could look something like this
```bash
_encryptions add "gpg"

_encrypt_gpg() {
}

_decrypt_gpg() {
}
```

You would also have to be able to extend the logic that determines the encryption method of an existing file.
Technically I guess it could try every decryption method until it succeeds.

**Comment by raspbeguy** [comment]
I think this would be a great idea. Maybe it could get inspiration from [pass](https://www.passwordstore.org/) encryption system.
Also, in a multiple users scenario, this system would enable multiple keys to access to the notes.
